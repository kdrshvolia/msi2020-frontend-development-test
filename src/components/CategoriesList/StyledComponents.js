import styled, { css } from 'styled-components';const CategoriesListWrapper = styled.div`  display: flex;`;const CategoryItem = styled.div`  height: 28px;  border: 2px solid #f8f8f8;  box-sizing: border-box;  border-radius: 6px;  font-family: Roboto;  font-style: normal;  font-weight: 500;  font-size: 12px;  letter-spacing: 2px;  text-transform: uppercase;  color: #ababab;  padding: 6px 15px;  cursor: pointer;`;export { CategoriesListWrapper, CategoryItem };