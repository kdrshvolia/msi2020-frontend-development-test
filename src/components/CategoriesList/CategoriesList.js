import React from 'react';
import PropTypes from 'prop-types';
import { CategoriesListWrapper, CategoryItem } from './StyledComponents';

const CategoriesList = ({ categories, setCategory }) => {
  return (
    <CategoriesListWrapper>
      {categories.length === 0
        ? null
        : categories.map((category) => (
            <CategoryItem onClick={setCategory}>{category.toUpperCase()}</CategoryItem>
          ))}
    </CategoriesListWrapper>
  );
};

CategoriesList.propTypes = {
  categories: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  setCategory: PropTypes.func.isRequired,
};

export default CategoriesList;
