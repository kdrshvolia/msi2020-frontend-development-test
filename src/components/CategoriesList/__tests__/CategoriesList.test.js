import React from 'react';
import renderer from 'react-test-renderer';
import CategoriesList from '../CategoriesList';

describe('CategoriesList', () => {
  it('renders correctly', () => {
    const categories = ['a', 'b'];
    const setCategory = jest.fn();
    const chosenCategory = 'a';
    const rendered = renderer.create(
      <CategoriesList
        categories={categories}
        setCategory={setCategory}
        chosenCategory={chosenCategory}
      />,
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
