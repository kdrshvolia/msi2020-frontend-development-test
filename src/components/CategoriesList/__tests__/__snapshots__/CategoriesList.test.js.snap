// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesList renders correctly 1`] = `
<div
  className="sc-AxjAm PLNtf"
>
  <div
    className="sc-AxirZ hqXbij"
    onClick={[Function]}
  >
    A
  </div>
  <div
    className="sc-AxirZ hCDbqu"
    onClick={[Function]}
  >
    B
  </div>
</div>
`;
